lib/carbon/storage.py

```diff
@@ -13,24 +13,84 @@
 limitations under the License."""
 
 import os
+import time
 import re
 import whisper
 
-from os.path import join, exists, sep
+from os.path import join, exists, sep, dirname
 from carbon.conf import OrderedConfigParser, settings
 from carbon.util import pickle
-from carbon import log
+from carbon import log, stats
 from carbon.exceptions import CarbonConfigException
+from collections import deque
+from twisted.internet import reactor
+from twisted.application.service import Service
 
 STORAGE_SCHEMAS_CONFIG = join(settings.CONF_DIR, 'storage-schemas.conf')
 STORAGE_AGGREGATION_CONFIG = join(settings.CONF_DIR, 'storage-aggregation.conf')
 STORAGE_LISTS_DIR = join(settings.CONF_DIR, 'lists')
 
+whisperCreates = deque()
 
 def getFilesystemPath(metric):
   metric_path = metric.replace('.', sep).lstrip(sep) + '.wsp'
   return join(settings.LOCAL_DATA_DIR, metric_path)
 
+def queueWhisperCreate(metric):
+  whisperCreates.append(metric)
+
+def createWhisperFile(metric, dbFilePath):
+    archiveConfig = None
+    xFilesFactor, aggregationMethod = None, None
+
+    for schema in SCHEMAS:
+      if schema.matches(metric):
+        log.creates('new metric %s matched schema %s' % (metric, schema.name))
+        archiveConfig = [archive.getTuple() for archive in schema.archives]
+        break
+
+    for schema in AGGREGATION_SCHEMAS:
+      if schema.matches(metric):
+        log.creates('new metric %s matched aggregation schema %s' % (metric, schema.name))
+        xFilesFactor, aggregationMethod = schema.archives
+        break
+
+    if not archiveConfig:
+      raise Exception("No storage schema matched the metric '%s', check your storage-schemas.conf file." % metric)
+
+    if not exists(dbFilePath):
```

Comment:
Per our chat, please add an `else` to this with instrumentation that counts the amount of times we've hit this condition but the file already existed.